torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -194,13 +203,10 @@ def from_float(
         zero_point_dtype: Optional[torch.dtype] = None,
         preserve_zero: bool = True,
         zero_point_domain: ZeroPointDomain = ZeroPointDomain.INT,
-        extended_layout: str = "plain",
-        # TODO: this is only for "tensor_core_tiled", need to figure out
-        # the proper API for this arg
-        inner_k_tiles: Optional[int] = None,
+        layout_type: LayoutType = PlainLayoutType(),
     ):
         original_shape = input_float.shape
-        if extended_layout == "tensor_core_tiled":
+        if isinstance(layout_type, TensorCoreTiledLayoutType):
```

Comment:
is it possible to move this inside the TensorCoreTiledLayout constructor

Reply:
oh, this is padding the input though, not sure how we can do that, but we can move the implementation to `TensorCoreTiledLayout` I think

Reply:
Yea I think that would be a bit cleaner if all layout specific code was kept within the layout class.